webapp/src/pages/registration/components/RegistrationComponent.js

```diff
@@ -0,0 +1,381 @@
+                            options={question.options}
+                            id={question.id}
+                            onChange={this.handleChangeDropdown}
+                            defaultValue={answer ? answer.label : ""}
+                            placeholder={answer ? this.getDropdownDescription(question.options, answer) : question.placeholder}
+                            label={question.description}
+                            required={question.is_required && !answer}
+                        />
+                    );
+                case FILE:
+                    return (
+                        <FormFileUpload
+                            Id={question.id}
+                            name={this.id}
+                            label={question.description}
+                            key={"i_" + key}
+                            value={answer}
+                            showError={validationError || this.state.uploadError}
+                            errorText={validationError || this.state.uploadError}
+                            uploading={this.state.uploading}
+                            uploadPercentComplete={this.state.uploadPercentComplete}
+                            uploadFile={this.handleUploadFile}
+                            uploaded={this.state.uploaded}
+                        />
+                    );
+                default:
+                    return (
+                        <p className="text-danger">
+                            WARNING: No control found for type {question.type}!
+          </p>
+                    );
+            }
+        };
+
+        if (isLoading) {
+            return (
+                <div class="d-flex justify-content-center">
+                    <div class="spinner-border" style={loadingStyle} role="status">
+                        <span class="sr-only">Loading...</span>
+                    </div>
+                </div>
+            );
+        }
+
+        if (error) {
+            return <div className={"alert alert-danger"}>{error}</div>;
+        }
+
+        return (
+            <div className="registration container-fluid pad-top-30-md">
+                <div className={this.state.formSucess ? "display-none" : "stretched"}>
+                    <h2>Registration</h2>
+                </div>
+                <div className={this.state.formSucess ? "card flat-card success stretched" : "display-none"}>
+                    <div >Successfully submitted</div>
+                    <div className="col-12">
+                        <button type="button"
+                            class="btn btn-primary pull-right"
+                            onClick={() => this.resetPage()}>Change your answers</button>
+                    </div>
+                </div>
+                <div className={this.state.formFailure ? "alert alert-danger stretched" : "display-none"}>
+                    <div>Something went wrong, please try again</div>
```

Comment:
Can you display the error message? I still live in hope that one day someone will include the error message when they email us instead of just saying "help me, it's not working" :)